### ISSUE TITLE
[patch] avoid freezes reading incomplete mp3 files

### ISSUE THREAD

**Issue by shaforostoff** [taglib/taglib#842]
avoid being stuck in previousFrameOffset() called from lastFrameOffset() when position >= length()

https://pastebin.com/xjc4PK5N

**Comment by scotchi** [comment]
Could you send an example file that's triggering this issue?  (to wheeler@kde.org)

**Comment by shaforostoff** [comment]
sorry, but i don't see any sense sending you problematic files. I have already sent you an aac file as part of another issue, but none of my submitted patches fixing rare corner cases are being accepted, with answers like, to quote, "Unless it's something very common, my impulse is not to work around this sort of brokenness."

**Comment by scotchi** [comment]
You're right that TagLib in general is intentionally rather strict with parsing files.  If there are broken files that aren't created by major applications (like, say, iTunes) we tend to not include code to work around them, and usually just ignore the rest of the metadata.  But it sounded like on this issue that it might lead to a hang in the application, which is a somewhat different case (and one that we definitely would want to work around).
